### ISSUE TITLE
Failing on certain versions of MacOS

### ISSUE THREAD

**Issue by stein-indee** [castlabs/electron-releases#34]
I am facing the following issue (seems identical to [this issue](https://github.com/castlabs/electron-releases/issues/30)) but on castlabs release v5) reported on certain macOS versions. Yosemite, El Capitan, Sierra and Mojave. Only High Sierra has had no failures reported so far (which is also the macOS version that I built and signed the app on, but that may be coincidental.) I have access to a Mojave system, and the problem is not reproducible there.

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000002, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Symbol not found: _kSecAttrKeyTypeECSECPrimeRandom
  Referenced from: /Applications/indee.app/Contents/MacOS/../Frameworks/Electron Framework.framework/Electron Framework
  Expected in: /System/Library/Frameworks/Security.framework/Versions/A/Security
 in /Applications/indee.app/Contents/MacOS/../Frameworks/Electron Framework.framework/Electron Framework

Is this an issue with the castlabs electron-fork? How do I resolve this? Thank you.

**Comment by khwaaj** [comment]
I can't reproduce this on Yosemite or Mojave, which is what I have immediately available. You are saying this happens with [v5.0.0-wvvmp-beta.3](https://github.com/castlabs/electron-releases/releases/tag/v5.0.0-wvvmp-beta.3), right?

The dump you provided looks like the same issue as the one you referenced, which was caused by problems with the availability macros of a certain version of the MacOSX10.12 SDK (causing non-weak linking of this symbol). The build machine has been updated with the fixed version so any build after v4.0.1 (and the pre-4.0 builds, since they are on a different SDK) should not have this problem. Either way this issue should not be seen on Sierra and later since the symbol in question should be available in `Security.framework`, so that you are seeing it on Sierra and Mojave is a bit strange.

Would you mind running the below on your application to verify that the symbol is in fact weakly linked?

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dyldinfo -bind indee.app/Contents/Frameworks/Electron\ Framework.framework/Electron\ Framework | grep PrimeRandom
```

You should see something like this (the `weak import` at the end being the important part):

```
__DATA  __got            0x060C1B38    pointer      0 Security         _kSecAttrKeyTypeECSECPrimeRandom (weak import)
```

**Comment by stein-indee** [comment]
When I execute that command (also tried with sudo in case that was throwing something off), I get the following result:

`dyldinfo failed: cannot open file in indee.app/Contents/Frameworks/Electron Framework.framework/Electron Framework`

What does this indicate?

**Comment by khwaaj** [comment]
You are probably just using the wrong path, what I provided above assumes you have a Terminal open in the directory of your application. After looking more closely at your dump above you should be able to just add `/Applications/` at the start of the path too `indee.app` to be able to run it from anywhere:

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dyldinfo -bind /Applications/indee.app/Contents/Frameworks/Electron\ Framework.framework/Electron\ Framework | grep PrimeRandom
```

**Comment by stein-indee** [comment]
Thank you. I think I understand what's going on.

I tried running the command with the correct path and it returned the following:
`__DATA  __got            0x060C1B38    pointer      0 Security         _kSecAttrKeyTypeECSECPrimeRandom`
This, as I understand, points to the issue you explained in your first post (non-weak linking).

After some digging, I realized that my build settings were not being correctly reflected in the electron-builder build process (specified version 5.0.0 in package.json but 4.00 was being used). I fixed this by uninstalling electron and reinstalling from the 5.0.0 release. After this, the build process correctly used 5.0.0. I verified this by re-running the command, which now returns this:
`__DATA  __got            0x060C1B38    pointer      0 Security         _kSecAttrKeyTypeECSECPrimeRandom (weak import)`
which hopefully implies this issue should be fixed.
I do not have access to the test machines right now but I will be able to test this across all major macOS versions by tomorrow morning and hopefully close this issue.

Thank you very much for your help. Have a great day!
